dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java

```diff
@@ -684,6 +685,9 @@ private void initTaskQueue() {
         if (processInstance.isComplementData() && complementListDate.size() == 0) {
             Map<String, String> cmdParam = JSONUtils.toMap(processInstance.getCommandParam());
             if (cmdParam != null && cmdParam.containsKey(CMDPARAM_COMPLEMENT_DATA_START_DATE)) {
```

Comment:
This is not just a complement. There should be a similar problem when you directly click Start

Reply:
> This is not just a complement. There should be a similar problem when you directly click Start

This part fixes that cmd param can overwrite global param when executing complement, change in ProcessService fix bug when directly click Start.

Reply:
I mean, it's not just a matter of complement. Can you help fix it